runtime/src/epoch_accounts_hash/manager.rs

```diff
@@ -0,0 +1,114 @@
+use {
+    super::EpochAccountsHash,
+    solana_sdk::{clock::Slot, hash::Hash},
+    std::sync::{Condvar, Mutex},
+};
+
+/// Manage the epoch accounts hash
+///
+/// Handles setting when an EAH calculation is requested and when it completes.  Also handles
+/// waiting for in-flight calculations to complete when the "stop" Bank must include it.
+#[derive(Debug)]
+pub struct Manager {
+    /// Current state of the epoch accounts hash
+    state: Mutex<State>,
+    /// This condition variable is used to wait for an in-flight EAH calculation to complete
+    cvar: Condvar,
+}
+
+impl Manager {
```

Comment:
any tests for the manager? It is being called from main code, I think, under normal conditions.